packages/loader/container-loader/src/container.ts

```diff
@@ -965,6 +965,7 @@ export class Container
 			this.subLogger,
 			this.storageAdapter,
 			offlineLoadEnabled,
+			async (storage, tree) => this.getDocumentAttributes(storage, tree),
```

Comment:
make getDocumentAttributes a free function, and then it can be used from anywhere without the need to pass it around. this could be done beforehand as a seperate pr

Reply:
good idea. I'll do that plus making fetchSnapshotCore and fetchSnapshotTree free functions as well